### PR TITLE
Compatibility with Firebase Functions

### PR DESCRIPTION
**Pull request by seiwonpark**
## Summary

* Node.js server POST request

    When requesting POST to Node.js server, and using `koa-bodyparser` package, I couldn't read body by `ctx.request.body` because of `body: string | Record<string, unknown>`. But when using previous code `body?: any`, then I could get request body by `ctx.request.body`.

* Firebase Functions POST request

  Firebase Functions already parses request (added links below) but still needs `koa-bodyparser` package to be installed. But it doesn't use `koa-bodyparser`. So unlike Node.js POST request, it cannot read request  body from `ctx.request.body` but by adding `body` property to `IncomingMessage`, it can read body by `ctx.req.body`.

<br/>   

## Checklist

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:      

    - [https://firebase.google.com/docs/functions/http-events#read_values_from_the_request](https://firebase.google.com/docs/functions/http-events#read_values_from_the_request)
    -  [https://stackoverflow.com/questions/58237305/post-request-hangs-timeout-when-trying-to-parse-request-body-running-koa-on-f](https://stackoverflow.com/questions/58237305/post-request-hangs-timeout-when-trying-to-parse-request-body-running-koa-on-f)
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Review by SimonSchick — CHANGES_REQUESTED** [review]
This should be done via type augmentations OR by writing separate type packages, environment specific augmentations should not be added to type packages.

**Comment by pkuczynski** [comment]
There is #53691 fixing this issue, waiting for approval...

**Comment by SeiwonPark** [comment]
> There is #53691 fixing this issue, waiting for approval...

Thank you @pkuczynski  for commenting! So type `any` doesn't seem to be safe at all. And I should not change those files.

Could I ask you one thing? About the `types/node/http.d.ts` changes that I tried to request ( [https://github.com/DefinitelyTyped/DefinitelyTyped/pull/54397/files](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/54397/files) ), would `Object(ctx.request.body)` be okay? I've typecasted the body.

Here's what I tried:   

```typescript
interface Payload {
    a: string
    b: string
}

export const fooFunc = async (ctx: Context, next: Next) => {
        const body: UserPayload = Object(ctx.request.body)
}
```

And it works. I could get body successfully. But I'm wondering if this typecasting would be okay (I mean if it's **safe** ). Thank you for sincere advice!

**Pull request closed without merging**
